subsquid/src/server-extension/resolvers/assets.ts

```diff
@@ -37,7 +37,19 @@ export class AssetsResolver {
 
     const manager = await this.tx();
 
-    const price = await getOrCreateHistoricalAssetPrice(manager, assetId, new Date(date || new Date()).getTime());
+    if (!date) {
+      const currentPrices = await getCurrentAssetPrices(manager);
+      const price =
+        currentPrices?.[assetId] ||
+        (await getOrCreateHistoricalAssetPrice(manager, assetId, new Date(date || new Date()).getTime()));
```

Comment:
date is null. just new Date().getTime() will be fine

Reply:
well spotted, carried it from a copy-paste 🙂